CIP-1694/README.md

```diff
@@ -0,0 +1,664 @@
+
+Votes can be cast multiple times for each governance action by a single key hash. Correctly submitted votes override any older votes for the same key hash and role.
+As soon as a governance action is ratified, voting ends. Any future votes are not considered or recorded (whether they are yes, no or abstain).
+
+#### Governance State
+
+When a governance action is successfully submitted to the chain, its progress will be tracked by the ledger state.
+In particular, the following will be tracked:
+
+* the governance action ID;
+* the epoch that the action expires;
+* the deposit amount;
+* the rewards address that will receive the deposit when it is returned;
+* the total yes/no/abstain votes of the Constitutional Committee for this action;
+* the total yes/no/abstain votes of the DReps for this action;
+* the total yes/no/abstain votes of the SPOs  for this action.
+
+#### Stale Votes
+
+The votes from the DReps and the SPOs may become meaningless after crossing an epoch boundary since they may become unregistered. Therefore, all unregistered votes are cancelled before any new votes are considered.
+
+#### Changes to the Stake Snapshot
+
+Since the stake snapshot changes at each epoch boundary, a new current voting tally must be calculated for each governance action based on the votes that have been cast before any new votes are counted.  This avoids the same stake being used two or more times.  _This new tally may result in immediate ratification of a governance action._
+
+### Delegated Representatives (DReps)
+
+> **Warning**
+> The design of the DReps is still subject to change and should not be conflated with Project Catalyst DReps.
+
+Existing stake keys will be given new responsibilities. They will be able to delegate their stake to DReps. DRep Registration will mimic the existing stake delegation mechanisms (via certificates).
+
+The following new types of voting certificates will be added:
+
+* DRep registration certificate. Including:
+  * DRep ID: hash of verification key
+  * a URL for any metadata that is relevant to the DRep
+  * a hash of the contents of this URL
+
+* DRep retirement certificate. Including:
+  * DRep ID
+  * epoch number after which the DRep will retire
+
+* Vote delegation certificate. Including:
+  * stake credential
+  * DRep ID
+
+// TODO: Provide CBOR specification in the annexe for those new certificates.
+
+The authorization scheme (i.e. which signatures are required) mimics the existing stake delegation certificate authorization scheme.
+
+#### New Stake Distribution for DReps
+
+In addition to the existing per-stake-credential distribution and the per-stake-pool distribution, we will now have a new per-DRep stake distribution.
+
+This distribution will determine how much stake is backed by each `yes` vote from a DRep. The exact stake snapshot used for block production by the SPOs will also be used for the DReps for voting.
```

Comment:
This would be a pretty big change to how we track stake, which already suffers from performance issues. Adding such a feature would drastically increase complexity and may make it much harder to have a smooth transition over the epoch boundary.

Reply:
As you say, this is not something we can stop. And indeed it is too drastic a change (we are piggy-backing off of the block production stake) for too little pay off.

Reply:
I already expected this to be somewhat problematic from a performance perspective. I don't have enough knowledge about Cardano's snapshotting process to evaluate the complexity of such a change.